java/arcs/core/data/Ttl.kt

```diff
@@ -24,15 +26,33 @@ sealed class Ttl(count: Int, val isInfinite: Boolean = false) {
             "must be either positive count on infinite, " +
                 "but got count=$count and isInfinite=$isInfinite"
         }
+        if (!isInfinite) {
+            requireNotNull(time) {
+                "time cannot be null for non infinite ttl"
+            }
+        }
     }
 
     override fun equals(other: Any?): Boolean =
         other is Ttl && minutes == other.minutes
 
     override fun hashCode(): Int = minutes.hashCode()
 
-    data class Minutes(val count: Int) : Ttl(count)
-    data class Hours(val count: Int) : Ttl(count)
-    data class Days(val count: Int) : Ttl(count)
-    object Infinite : Ttl(-1, true)
+    @Suppress("")
+    fun calculateExpiration(): Long {
+        if (isInfinite) {
+            return RawEntity.NO_EXPIRATION
+        } else {
+            return requireNotNull(time).currentTimeMillis.plus(minutes * 60 * 1000)
+        }
+    }
```

Comment:
Could lift the return, and make this a function expression:

```kotlin
fun calculateExpiration(): Long = 
    if (isInfinite) RawEntity.NO_EXPIRATION 
    else requireNotNull(time).currentTimeMillis + (minutes * 60 * 1000)
```

Reply:
done, thanks!